lfs_ffi.lua

```diff
@@ -999,30 +1238,28 @@ if OS == 'Windows' then
         return nil
     end
 else
-    ffi.cdef('unsigned long readlink(const char *path, char *buf, size_t bufsize);')
+    ffi.cdef('ssize_t readlink(const char *path, char *buf, size_t bufsize);')
     function get_link_target_path(link_path)
-        local size = MAXPATH
-        while true do
-            local buf = ffi.new('char[?]', size)
-            local read = lib.readlink(link_path, buf, size)
-            if read == -1 then
-                return nil, errno()
-            end
-            if read < size then
-                return ffi_str(buf)
-            end
-            size = size * 2
+        local statbuf = ffi.new(stat_type)
+        if lstat_func(link_path, statbuf) == -1 then
+            error(errno(),2)
+        end
+        local size = statbuf.st_size
+        local buf = ffi.new('char[?]', size + 1)
+        local read = lib.readlink(link_path, buf, size)
+        if read == -1 then
+            --einval when link_path is not a link
+            if ffi.errno()~=22 then error(errno(),2) end
+            return nil, errno()
         end
+        if read > size then
+            error("not enought size for readlink "..errno(),2)
```

Comment:
Ditto

Reply:
Sorry, I dont understand the last comment.

Reply:
size is now provided by lstat and if results size==0 then size=MAXPATH
In any case read should be read<=size and so the error is thrown

Reply:
@sonoro1234
It is my bad not to comment it clearly...
I mean, it would be better to use `return nil, "not enought size for readlink"` instead of raising an error.

Reply:
But second return argument is not used, so the behaviour is the same as if it was not a link and the error is lost!!

Reply:
@sonoro1234
So it is important to check the return value of `get_link_target_path`. It's my bad to forget checking the return value which needs to be fixed.

Reply:
You are right, I saw lfs.c and made a new commit to emulate behaviour